### PR TITLE
feat: add Create / Update / Delete requests

### PR DESCRIPTION
**Pull request by kalexmills**
WIP For #388 

Based on the APIs in the ticket.

Thanks to the maintainers for having such a well-factored project. This has been easy so far.

### CODE REVIEW

**Comment by kalexmills** [comment]
Only thing I'm waiting on is actually testing this against GitHub via Ammonite (which I will do soon), since it seems the integration tests are only used / needed for reads.

**Comment by kalexmills** [comment]
Huh... The doc build must be using a separate coverage tool from this GitBot. The CI process is claiming coverage is at less than 70% and failing the builds because of it. :-/

**Comment by kalexmills** [comment]
Tested this and it seems to be working, so I'm marking as ready for review.

I did find that requiring the client to convert to base64 is annoying, so the API now takes an array of bytes.

Only issue with this is it does require keeping the entire file in memory, but that's GitHub's API. I don't see how we could do a light memory footprint without streaming JSON which would require deeper changes.

**Comment by kalexmills** [comment]
One more tweak to edit the docs.

File: github4s/src/main/scala/github4s/interpreters/RepositoriesInterpreter.scala [hunk 1]
```diff
@@ -23,6 +23,7 @@ import github4s.GithubResponses.GHResponse
 import github4s.domain._
 import github4s.Decoders._
 import github4s.Encoders._
+import com.github.marklister.base64.Base64._
```
[the remark this answers is not in the document]
Comment (kalexmills) [reply github4s/src/main/scala/github4s/interpreters/RepositoriesInterpreter.scala]: It's used in a few other places as well.

**Comment by kalexmills** [comment]
There are issues with relying on the `Commit` object which uses some custom decoding that powers through multiple layers of JSON at once. After further testing, I've noticed that object is only used in `ListCommit`.

Users could see a JsonParseException if they try to use it.

Anyway the `Commit` object in `Repository.scala` is inappropriate for this endpoint, so I'll need to make some changes.

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.23.1-SNAPSHOT"
\ No newline at end of file
+version in ThisBuild := "0.24.0-SNAPSHOT"
```
Comment (BenFradet) [inline version.sbt]: please don't change the version as part of this PR.

File: github4s/src/main/scala/github4s/interpreters/RepositoriesInterpreter.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenFradet) [reply github4s/src/main/scala/github4s/interpreters/RepositoriesInterpreter.scala]: yes might be a good idea 👍

File: version.sbt (hunk 2 above)
Comment (kalexmills) [reply version.sbt]: Woops! did not mean to push that, haha. Just using it locally.

**Comment by kalexmills** [comment]
@fedefernandez I do not think 'create or update' can function as an upsert. The update [requires that you include the `sha` as a parameter](https://developer.github.com/v3/repos/contents/#create-or-update-a-file), so the user has to know whether the file exists or not.

**Comment by kalexmills** [comment]
@fedefernandez Thanks! Done.

**Pull request merged**